### ISSUE TITLE
RTP extension no longer works

### ISSUE THREAD

**Issue by expipiplus1** [neoclide/coc.nvim#2282]
## Result from CocInfo

failing:

```
2020-08-27T11:08:17.776 DEBUG (pid:10586) [workspace] - buffer created 1
2020-08-27T11:08:17.777 DEBUG (pid:10586) [events] - Event: BufEnter [ 1 ]
2020-08-27T11:08:17.778 DEBUG (pid:10586) [events] - Event: BufWinEnter [ 1, 1000 ]
2020-08-27T11:08:17.824 INFO (pid:10586) [plugin] - coc.nvim 0.0.78-eb361c5b3e initialized with node: v10.19.0 after 75ms
2020-08-27T11:08:17.929 DEBUG (pid:10586) [events] - Event: CursorHold [ 1 ]
```

succeeding:

```
2020-08-27T11:10:04.213 DEBUG (pid:12003) [workspace] - buffer created 1
2020-08-27T11:10:04.214 DEBUG (pid:12003) [events] - Event: BufEnter [ 1 ]
2020-08-27T11:10:04.219 DEBUG (pid:12003) [events] - Event: BufWinEnter [ 1, 1000 ]
2020-08-27T11:10:04.279 INFO (pid:12003) [language-client-index] - Language server "Haskell (examples)" started with 12050
2020-08-27T11:10:04.282 INFO (pid:12003) [plugin] - coc 0.0.78-6f1ec22f95 initialized with node: v10.19.0
2020-08-27T11:10:04.312 DEBUG (pid:12003) [languages] - created service source Haskell (examples)
2020-08-27T11:10:04.688 DEBUG (pid:12003) [events] - Event: CursorHold [ 1 ]
```

## Describe the bug

An extension supplied via RTP no longer functions

## Reproduce the bug

- Set runtimepath: `set runtimepath^=/some/path`
  - `/some/path` contains `package.json` and `dist/extension.js` *I think that this is what needs to change, but I don't know what to*
- Build coc at `eb361c5b3eddc6c41c8a2e491615b5da6a8b4aff` and observe that the plugin in RTP isn't loaded
- Build coc at `eb361c5b3eddc6c41c8a2e491615b5da6a8b4aff^` and observe that the plugin works

**Comment by fannheyward** [comment]
Can't reproduce.

Make sure your `package.json` main filed points to your main js file.

I can't find commit `eb361c5b3eddc6c41c8a2e491615b5da6a8b4aff` in coc.

**Comment by expipiplus1** [comment]
Ah, sorry I had another (unrelated change). The hash is `14623e1ca36254fc9271c3bf20a60fe92aa2f0bd`

package.json has `"main": "./dist/extension.js"`

**Comment by expipiplus1** [comment]
Ah, I do see the extension in `CocList extensions`. However when I try to disable then reenable it I get 

```
[coc.nvim] error: Error: haskell not installed as global extension.
    at Extensions.toggleExtension (/nix/store/djis2g6h3jbcyzdksp1b1ffhl3awck2n-vimplugin-coc-nvim-2020-01-05/share/vim-plugins/coc-nvim/build/index.js:40654:23)
    at Object.ExtensionList.addAction [as execute] (/nix/store/djis2g6h3jbcyzdksp1b1ffhl3awck2n-vimplugin-coc-nvim-2020-01-05/share/vim-plugins/coc-nvim/build/index.js:66582:44)
    at Promise.all.items.map.item (/nix/store/djis2g6h3jbcyzdksp1b1ffhl3awck2n-vimplugin-coc-nvim-2020-01-05/share/vim-plugins/coc-nvim/build/index.js:61585:76)
    at Array.map (<anonymous>)
    at ListManager.doItemAction (/nix/store/djis2g6h3jbcyzdksp1b1ffhl3awck2n-vimplugin-coc-nvim-2020-01-05/share/vim-plugins/coc-nvim/build/index.js:61585:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

I guess that it's because this extension was build against coc.nvim 0.77 and 14623e1ca36254fc9271c3bf20a60fe92aa2f0bd is too new?

**Comment by expipiplus1** [comment]
Happy to leave this closed until I can test 0.0.78 (or 79) when they're released on npm
